src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -167,81 +164,7 @@ public async Task<ContentItem> GetAsync(string contentItemId, VersionOptions opt
-                    .Where(x => x.ContentItemId == contentItemId && x.Latest == true)
-                    .FirstOrDefaultAsync();
-            }
-            else if (options.IsDraft && !options.IsDraftRequired)
-            {
-                contentItem = await _session
-                    .Query<ContentItem, ContentItemIndex>()
-                    .Where(x =>
-                        x.ContentItemId == contentItemId &&
-                        x.Published == false &&
-                        x.Latest == true)
-                    .FirstOrDefaultAsync();
-            }
-            else if (options.IsDraft || options.IsDraftRequired)
-            {
-                // Loaded whatever is the latest as it will be cloned
-                contentItem = await _session
-                    .Query<ContentItem, ContentItemIndex>()
-                    .Where(x =>
-                        x.ContentItemId == contentItemId &&
-                        x.Latest == true)
-                    .FirstOrDefaultAsync();
-            }
-            else if (options.IsPublished)
-            {
-                // If the published version is requested and is already loaded, we can
-                // return it right away
-                if (_contentManagerSession.RecallPublishedItemId(contentItemId, out contentItem))
-                {
-                    return contentItem;
-                }
-
-                contentItem = await _session.ExecuteQuery(new PublishedContentItemById(contentItemId)).FirstOrDefaultAsync();
-            }
-
-            if (contentItem == null)
-            {
-                return null;
-            }
-
-            contentItem = await LoadAsync(contentItem);
-
-            if (options.IsDraftRequired)
-            {
-                // When draft is required and latest is published a new version is added
-                if (contentItem.Published)
-                {
-                    // We save the previous version further because this call might do a session query.
-                    var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
-
-                    // Check if not versionable, meaning we use only one version
-                    if (contentTypeDefinition != null && !contentTypeDefinition.IsVersionable())
-                    {
-                        contentItem.Published = false;
-                    }
-                    else
-                    {
-                        // Save the previous version
-                        _session.Save(contentItem, checkConcurrency: true);
-
-                        contentItem = await BuildNewVersionAsync(contentItem);
-                    }
-                }
-
-                // Save the new version
-                _session.Save(contentItem, checkConcurrency: true);
-            }
-
-            return contentItem;
+            return (await GetAsync(new[] { contentItemId }, options)).FirstOrDefault();
```

Comment:
Not sure it's a good pattern to use inside the `GetAsync()` returning one item, the same `GetAsync()` returning multiple items. Maybe good in term of refactoring, but not for performance, by using the 2nd version we allocate and use more collections (even if they are single item collections).

For example, to return a single published item, here we allocate an `Array`, and in the `GetAsync()` returning multiple items, as I can see we allocate 3 `List`s, and then here an additional Linq `FirstOrDefault()`. Also the query itself using `IsIn(ids)` may be less efficient, also in the single item version we were using `PublishedContentItemById : ICompiledQuery<ContentItem>` which is a pre-compiled query.